core/tests/webdriverio_utils/BlogPages.js

```diff
@@ -0,0 +1,308 @@
+    for (i = 0; i < keywords.length; i++) {
+      // Here Newline Character is used as ENTER KEY.
+      await action.setValue(
+        'Blog Post Tag filter dropdown ',
+        tagFilterInput,
+        keywords[i] + '\n'
+      );
+    }
+  };
+
+  this.navigateToBlogPostPage = async function(title) {
+    await this.waitForBlogPostsToLoad();
+    var blogPostTiles = await blogPostTilesSelector();
+    for (i = 0; i < blogPostTiles.length; i++) {
+      var blogPostTile = blogPostTiles[i];
+      var blogPostTitleContainer = await blogPostTile.$(
+        '.e2e-test-blog-post-tile-title');
+      // The element is not interactable when we call getText(), so it returns
+      // null. To avoid that we are waiting till the element becomes clicakble
+      // as we do not have any alternative for checking interactibility.
+      await waitFor.elementToBeClickable(
+        blogPostTitleContainer, 'Blog Post title is not interactable');
+      var blogPostTitle = await action.getText(
+        `Blog Post Tile Title ${i}`, blogPostTitleContainer);
+      if (blogPostTitle === title) {
+        await action.click('blog post tile', blogPostTile);
+        await waitFor.pageToFullyLoad();
+        break;
+      }
+    }
+  };
+
+  this.expectBlogPostPageTitleToBe = async function(title) {
+    await waitFor.visibilityOf(
+      oppiaBlogPostPageContainer,
+      'Oppia Blog Post Page taking too long to display');
+    await waitFor.visibilityOf(
+      blogPostTitleContainer,
+      'Oppia Blog Post Page taking too long to display');
+    var blogPostTitle = await action.getText(
+      'Blog Post Title', blogPostTitleContainer);
+    expect(blogPostTitle).toEqual(title);
+  };
+
+  this.navigateToBlogHomePageWithBackButton = async function() {
+    await action.click(
+      'button to navigate back to blog home page',
+      navigateToBlogHomePageButton
+    );
+    await waitFor.pageToFullyLoad();
+    await this.waitForVisibilityOfBlogHomePageContainer();
+  };
+
+  this.navigateToBlogAuthorPage = async function() {
+    // The element is not interactable when we call getText(), so it returns
+    // null. To avoid that we are waiting till the element becomes clicakble
+    // as we do not have any alternative for checking interactibility.
+    await waitFor.elementToBeClickable(
+      authorProfilePhoto, 'Author Profile photo is not interactable');
+    await action.click('author profile photo', authorProfilePhoto);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.expectNumberOfRecommendationPostsToBe = async function(number) {
+    if (number) {
+      await this.waitForBlogPostsToLoad();
+    } else {
+      expect(await blogPostsList.isDisplayed()).toBe(false);
+    }
+    var blogPostRecommendationTiles = (
+      await blogPostRecommendationTilesSelector());
+    expect(blogPostRecommendationTiles.length).toBe(number);
+  };
+
+  this.expectRecommendationToContainPostWithTitle = async function(title) {
+    if (title) {
```

Comment:
In what scenario would this be falsy? From the function name it seems like a title should always be provided

Reply:
Removed. Title will be always provided

Reply:
I still see this in the diff

Reply:
Somehow got reverted. Done.